### ISSUE TITLE
BUG: Parquet roundtrip fails with numerical categorical dtype

### ISSUE THREAD

**Issue by adrienpacifico** [pandas-dev/pandas#60491]
### Pandas version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pandas.pydata.org/docs/whatsnew/index.html) of pandas.

- [ ] I have confirmed this bug exists on the [main branch](https://pandas.pydata.org/docs/dev/getting_started/install.html#installing-the-development-version-of-pandas) of pandas.

### Reproducible Example

```python
>>> df=pd.DataFrame({"A": [1, 2, 3], "B": [4, 5, 6]})
>>> df = df.astype({'A':'category'})
>>> print(df.dtypes)
A    category
B       int64
dtype: object
>>> df.to_parquet('test.parquet')
>>> df_roundtrip = pd.read_parquet('test.parquet')
>>> print(df_roundtrip.dtypes)
A    int64
B    int64
dtype: object
>>> assert df_roundtrip.dtypes.equals(df.dtypes)

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AssertionError
```

### Issue Description

Roundtrip does not work.

### Expected Behavior

df_roundtrip has the same dtypes as df.dtypes

### Hot-Fix
```python
df=pd.DataFrame({"A": [1, 2, 3], "B": [4, 5, 6]})
df = df.astype({'A':'str'}).astype({'A':'category'})
print(df.dtypes)
df.to_parquet('test.parquet')
df_roundtrip = pd.read_parquet('test.parquet')
print(df_roundtrip.dtypes)

assert df_roundtrip.dtypes.equals(df.dtypes)
```

### Installed Versions

<details>

INSTALLED VERSIONS
------------------
commit                : 0691c5cf90477d3503834d983f69350f250a6ff7
python                : 3.12.2
python-bits           : 64
OS                    : Darwin
OS-release            : 23.5.0
Version               : Darwin Kernel Version 23.5.0: Wed May  1 20:16:51 PDT 2024; root:xnu-10063.121.3~5/RELEASE_ARM64_T8103
machine               : arm64
processor             : arm
byteorder             : little
LC_ALL                : None
LANG                  : fr_FR.UTF-8
LOCALE                : fr_FR.UTF-8

pandas                : 2.2.3
numpy                 : 2.0.2
pytz                  : 2024.2
dateutil              : 2.9.0.post0
pip                   : 24.0
Cython                : None
sphinx                : None
IPython               : 8.24.0
adbc-driver-postgresql: None
adbc-driver-sqlite    : None
bs4                   : 4.12.3
blosc                 : None
bottleneck            : 1.4.2
dataframe-api-compat  : None
fastparquet           : 2024.11.0
fsspec                : 2024.10.0
html5lib              : 1.1
hypothesis            : 6.122.1
gcsfs                 : 2024.10.0
jinja2                : 3.1.4
lxml.etree            : 5.3.0
matplotlib            : 3.9.3
numba                 : 0.60.0
numexpr               : 2.10.2
odfpy                 : None
openpyxl              : 3.1.5
pandas_gbq            : 0.24.0
psycopg2              : 2.9.10
pymysql               : 1.4.6
pyarrow               : 18.1.0
pyreadstat            : 1.2.8
pytest                : 8.3.4
python-calamine       : None
pyxlsb                : 1.0.10
s3fs                  : 2024.10.0
scipy                 : 1.14.1
sqlalchemy            : 2.0.36
tables                : 3.10.1
tabulate              : 0.9.0
xarray                : 2024.11.0
xlrd                  : 2.0.1
xlsxwriter            : 3.2.0
zstandard             : 0.23.0
tzdata                : 2024.2
qtpy                  : 2.4.2
pyqt5                 : None

</details>

**Comment by ppsmoraes** [comment]
I've tried using the fastparquet engine, and it seems to work. Whatever the problem is, it lies with the way the pyarrow engine reads the Parquet file.

Here is the code example:
``` python
import pandas as pd

df = pd.DataFrame({"A": [1, 2, 3], "B": [4, 5, 6]})
df = df.astype({'A':'category'})
print(df.dtypes)
# A    category
# B       int64
# dtype: object
df.to_parquet('test.parquet')

df_roundtrip = pd.read_parquet('test.parquet')
print(df_roundtrip.dtypes)
# dtype: object
# A    int64
# B    int64

df_roundtrip_fp = pd.read_parquet('test.parquet', engine='fastparquet')
print(df_roundtrip_fp.dtypes)
# A    category
# B       int64
# dtype: object

result = df_roundtrip.equals(df)
print(result)
# False

result = df_roundtrip_fp.equals(df)
print(result)
# True
```

**Comment by rhshadrach** [comment]
Agreed categorical-integers (and other dtypes) should roundtrip alongside categorial-strings. Further investigations and PRs to fix are welcome!

**Comment by tsafacjo** [comment]
can I take it ?

**Comment by myenugula** [comment]
take

**Comment by myenugula** [comment]
Seems that the core issue is within Pyarrow, not Pandas. 

When a DataFrame with a numeric categorical column is saved to Parquet and read back using the PyArrow engine (the default), the categorical dtype is lost and the column is converted back to a regular numeric type.
The issue is in how PyArrow handles dictionary-encoded columns during Parquet roundtrips. PyArrow recognizes dictionary-encoded string columns as categorical, but doesn't preserve the categorical information for numeric dictionary-encoded columns when reading from Parquet.

The key behavior difference is in how the PyArrow engine handles dictionary encoding during read:
1. For string categoricals: Dictionary encoding is preserved as categorical dtype
2. For numeric categoricals: Dictionary encoding is unpacked to numeric type, losing categorical metadata
